### PR TITLE
Fix function-handling in CSSDecl parser; add CSSRewriteURL parser

### PR DESCRIPTION
**Pull request by mdittmer**
CSSRewrite parser provides methods to replace with data URLs. Rolled in with fix for function-handling because RewriteURL rules depend on changed structures in Decl.

### CODE REVIEW

File: js/foam/grammars/CSSFont.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  requires: [
+    'XHR',
+    'foam.util.Base64Encoder',
+  ],
+  imports: [
+    'assert',
+    'document',
+  ],
+
+  documentation: function() {/* A <b>very permissive</b> ASCII CSS parser that
```
Comment (kgrgreer) [inline js/foam/grammars/CSSFont.js]: This description doesn't seem to correspond with the model name.

Comment (mdittmer) [inline js/foam/grammars/CSSFont.js]: Correct. Copy-paste error. Will fix.

On 25 November 2015 at 10:10, Kevin Glen Roy Greer <notifications@github.com

File: js/foam/grammars/CSSRewriteURL.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  documentation: function() {/* A slightly modified
+    $$DOC{ref:'foam.grammars.CSSDecl'} prefix-rewrite parser that also knows how
+    to expand remote URLs to dataURLs after fetching the data.
+  */},
+
+  constants: {
+    PREFIXES: [
+      '-webkit-'
+    ],
+    PREFIXED_KEYS: {
```
Comment (kgrgreer) [inline js/foam/grammars/CSSRewriteURL.js]: Why are these duplicated?

Comment (mdittmer) [inline js/foam/grammars/CSSRewriteURL.js]: They shouldn't be. Fixed in latest diff.

On 25 November 2015 at 15:14, Kevin Glen Roy Greer <notifications@github.com

File: js/foam/grammars/CSSRewriteURL.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      name: 'parser',
+      lazyFactory: function() {
+        return SkipGrammar.create(this.parser_, seq('/*', repeat(not('*/', anyChar)), '*/'));
+      },
+    },
+    {
+      name: 'parser_',
+      lazyFactory: function() {
+        var css = this;
+        var r = function() { return str(repeat.apply(this, arguments)); };
```
Comment (kgrgreer) [inline js/foam/grammars/CSSRewriteURL.js]: I think it's simpler to just use str(repeat(...)) or str(plus(...)) than for someone to have to figure out what you're doing here and then later in the grammar.

Comment (mdittmer) [inline js/foam/grammars/CSSRewriteURL.js]: Very well. Shall I remove the same pattern in CSSDecl as well?

On 25 November 2015 at 15:16, Kevin Glen Roy Greer <notifications@github.com

Comment (kgrgreer) [inline js/foam/grammars/CSSRewriteURL.js]: Yes.

Comment (mdittmer) [inline js/foam/grammars/CSSRewriteURL.js]: Fixed in both (see latest diff).

On 25 November 2015 at 17:48, Kevin Glen Roy Greer <notifications@github.com

**Pull request merged**